src/ui/components/NetworkMonitor/useCopyAsCURL.ts

```diff
@@ -0,0 +1,78 @@
+import { RequestBodyData } from "@replayio/protocol";
+import { useState } from "react";
+import { Row } from "react-table";
+
+import { fetchRequestBody } from "ui/actions/network";
+import { getRequestBodyById } from "ui/reducers/network";
+import { useAppDispatch, useAppSelector, useAppStore } from "ui/setup/hooks";
+
+import {
+  BodyPartsToUInt8Array,
+  RawToImageMaybe,
+  RawToUTF8,
+  URLEncodedToPlaintext,
+} from "./content";
+import { RequestSummary, findHeader } from "./utils";
+
+type State = "ready" | "loading" | "complete" | "error";
+
+export default function useCopyAsCURL(
+  row: Row<RequestSummary>,
+  resetAfterDelay: number = 2_500
+): {
+  copy: () => void;
+  state: State;
+} {
+  const dispatch = useAppDispatch();
+  const store = useAppStore();
+
+  let requestBody = useAppSelector(state => getRequestBodyById(state, row.id));
+
+  const [state, setState] = useState<State>("ready");
+
+  const copy = async () => {
+    try {
+      if (row.original.hasRequestBody && requestBody == null) {
+        setState("loading");
+
+        await dispatch(fetchRequestBody(row.id));
+
+        const state = store.getState();
+        requestBody = getRequestBodyById(state, row.id);
```

Comment:
@markerikson Does this fall within idiomatic Redux patterns or do you have an alternate suggestion?